src/PdfReader/index.tsx

```diff
@@ -30,26 +33,48 @@ function pdfReducer(state: PdfState, action: PdfReaderAction): PdfState {
         ...state,
         isScrolling: action.isScrolling,
       };
+
+    case 'SET_RESOURCE':
+      return {
+        ...state,
+        resource: action.resource,
+      };
+
+    case 'SET_PAGENUM':
+      return {
+        ...state,
+        pageNumber: action.pageNum,
+      };
   }
 }
 
 export default function usePdfReader(args: ReaderArguments): ReaderReturn {
   const { webpubManifestUrl, manifest } = args ?? {};
-  const [state] = React.useReducer(pdfReducer, {
+
+  const [state, dispatch] = React.useReducer(pdfReducer, {
     type: 'PDF',
     colorMode: 'day',
     isScrolling: false,
     fontSize: 16,
     fontFamily: 'sans-serif',
+    resource: '',
+    pageNumber: 1,
   });
 
+  const [numPages, setNumPages] = useState(0);
+  const [isLoading, setLoading] = useState(true);
```

Comment:
I think you can derive the `isLoading` state based on whether `resource` is `null` in the state. That's what I would do. I also think I would store the `numPages` in the state with everything else. I don't think you should need `useState` at all since we have `useReducer`